frontend/src/components/assessment-creation/FilterMenu.tsx

```diff
@@ -1,12 +1,132 @@
-import React from "react";
-import { Button } from "@chakra-ui/react";
+import React, { SetStateAction } from "react";
+import {
+  Button,
+  Text,
+  Popover,
+  PopoverBody,
+  PopoverContent,
+  PopoverFooter,
+  PopoverTrigger,
+  Divider,
+  Flex,
+} from "@chakra-ui/react";
+import { Select } from "chakra-react-select";
 import { FunnelIcon } from "../../assets/icons";
 
-const FilterMenu = (): React.ReactElement => {
+export type Option = {
+  value: string;
+  label: string;
+};
+
+export type FilterMenuProp = {
+  label: string;
+  setState: React.Dispatch<SetStateAction<Array<string>>>;
+  options: Option[];
+};
+
+interface FilterMenuProps {
+  filterProps: FilterMenuProp[];
+}
+
+function AddFilter(
+  optionsSelected: { [key: string]: string[] },
+  filterProps: FilterMenuProp[],
+): void {
+  filterProps.forEach((prop: FilterMenuProp) => {
+    if (
+      optionsSelected[prop.label] &&
+      optionsSelected[prop.label].length !== 0
+    ) {
+      prop.setState(optionsSelected[prop.label]);
+    } else {
+      prop.setState([]);
+    }
+  });
+}
+
+const FilterMenu = ({ filterProps }: FilterMenuProps): React.ReactElement => {
+  const optionsSelected = {} as { [key: string]: string[] };
+  const attributeList = filterProps.map((filterProp, i) => (
+    <>
+      <Flex pt="2%" px={3} pb={4} justifyContent="space-around">
+        <div style={{ width: "7rem" }}>
+          <Text color="blue.300" textStyle="link">
+            {filterProp.label}
+          </Text>
+        </div>
+
+        <div style={{ width: "23rem" }}>
+          <Select
+            chakraStyles={{
+              dropdownIndicator: (provided) => ({
+                ...provided,
+                background: "white",
+              }),
+              downChevron: (provided) => ({
+                ...provided,
+                boxSize: "1.5rem",
+              }),
+              crossIcon: (provided) => ({
+                ...provided,
+                boxSize: "0.8rem",
+              }),
+            }}
+            focusBorderColor="blue.300"
+            onChange={(choices) => {
+              optionsSelected[filterProp.label] = choices.map(
+                (choice) => choice.value,
+              );
+            }}
+            isMulti
+            name="colors"
+            options={filterProp.options}
+          />
+        </div>
+      </Flex>
+      {i !== filterProps.length - 1 && <Divider ml={2} mb={2} maxWidth="95%" />}
+    </>
+  ));
   return (
-    <Button minWidth="5%" leftIcon={<FunnelIcon />} variant="tertiary">
-      Filter
-    </Button>
+    <>
```

Comment:
can remove this empty tag

Reply:
That empty tag is necessary since it wraps both Popover Trigger and Popover Content, it's also there within the SortMenu component

Reply:
but i think the most outer empty tag can be removed?